restapi/src/main/java/io/stargate/web/docsapi/dao/DocumentDB.java

```diff
@@ -126,9 +129,23 @@ public void maybeCreateTable(String keyspaceName, String tableName) {
                   String.join(", ", allPathColumnNames)))
           .get();
 
-      dataStore
-          .query(String.format("CREATE INDEX ON \"%s\".\"%s\"(leaf)", keyspaceName, tableName))
-          .get();
+      // After creating the table, it can take up to 2 seconds for permissions cache to be updated,
```

Comment:
Would it make sense to introduce a new method into `persistence-api` to allow waiting for this intelligently or perhaps to receive a notification of some sort?

Reply:
yes, that's a really good point... exposing some method that creates a table and blocks until permissions are updated would be really good

Reply:
I have no idea how I'd do that though ^_^

Reply:
Let's file an issue :)

Reply:
Upon looking into this, I could use an implementation of QueryInterceptor to intercept a Create Table statement and then sleep for a bit...that still does a sleep though which is really not preferable.

I think maybe the key to this would be having closer control over the perms cache like Jeremiah mentioned is available in DSE...however, that probably will not work in C* 3.11/4, which is not so great...

For now, I decided to force a re-login to force refreshing of permissions.  I believe this will work across all three versions we care about.